tendrl/commons/utils/log_utils.py

```diff
@@ -0,0 +1,31 @@
+""" Handles logging functionality. """
+import sys
+from tendrl.commons.event import Event
+from tendrl.commons.message import Message
+
+
+def log(log_priority,publisher_id,**log_payload):
```

Comment:
Here the problem is message framework is identifying caller, line number, function names from stack, so it will give wrong info for that fields, Better identify those details inside this new function and pass to the message framework.  And don't for get to delete the code for find caller details in message framework.  

caller = getframeinfo(stack()[1][0])
 self.caller = {"filename": caller.filename,
                      "line_no": caller.lineno,
                       "function": caller.function}
add these lines in your function and pass these details to message framework, and remove these code from message framwork.

please check the parameters, i this it will not work for job_updates, you have to add job_id, flow_id, cluster_id, parent these details also in function parameter.   

use parameter like 
 def __init__(self, priority, publisher, payload, job_id=None,
                 flow_id=None, parent_id=None, cluster_id=None,
                 message_id=None, timestamp=None, node_id=None,
                 caller=None)